junifer/api/tests/test_functions.py

```diff
@@ -144,10 +147,294 @@ def test_run_and_collect(tmp_path: Path) -> None:
+        The path to the test directory.
+    monkeypatch : pytest.MonkeyPatch
+        The monkeypatch object.
+
+    """
+    with pytest.raises(ValueError, match="Either delete the directory"):
+        with monkeypatch.context() as m:
+            m.chdir(tmp_path)
+            # First generate assets
+            queue(
+                config={"elements": ["sub-001"]},
+                kind="HTCondor",
+                jobname="prevent_overwrite",
+            )
+            # Re-run to trigger error
+            queue(
+                config={"elements": ["sub-001"]},
+                kind="HTCondor",
+                jobname="prevent_overwrite",
+            )
+
+
+def test_queue_assets_allow_overwrite(
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test overwriting of queue files.
+
+    Parameters
+    ----------
+    tmp_path : pathlib.Path
+        The path to the test directory.
+    monkeypatch : pytest.MonkeyPatch
+        The monkeypatch object.
+    caplog : pytest.LogCaptureFixture
+        The logcapturefixture object.
+
+    """
+    with monkeypatch.context() as m:
+        m.chdir(tmp_path)
+        # First generate assets
+        queue(
+            config={"elements": ["sub-001"]},
+            kind="HTCondor",
+            jobname="allow_overwrite",
+        )
+        with caplog.at_level(logging.INFO):
+            # Re-run to overwrite
+            queue(
+                config={"elements": ["sub-001"]},
+                kind="HTCondor",
+                jobname="allow_overwrite",
+                overwrite=True,
+            )
+            assert "Deleting existing job directory" in caplog.text
+
+
+@pytest.mark.parametrize(
+    "with_",
+    [
+        "a.py",
+        ["a.py"],
+        ["a.py", "b"],
+    ],
+)
+def test_queue_with_imports(
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+    caplog: pytest.LogCaptureFixture,
+    with_: Union[str, List[str]],
+) -> None:
+    """Test queue with `with` imports.
+
+    Parameters
+    ----------
+    tmp_path : pathlib.Path
+        The path to the test directory.
+    monkeypatch : pytest.MonkeyPatch
+        The monkeypatch object.
+    caplog : pytest.LogCaptureFixture
+        The logcapturefixture object.
+    with_ : str or list of str
+        The parametrized imports.
+
+    """
+    with monkeypatch.context() as m:
+        m.chdir(tmp_path)
+        # Create test file, keeping it simple without conditionals
+        (tmp_path / "a.py").touch()
+        with caplog.at_level(logging.DEBUG):
+            queue(
+                config={"with": with_},
+                kind="HTCondor",
+                elements="sub-001",
+            )
+            assert "Copying" in caplog.text
+            assert "Queue done" in caplog.text
+
+
+@pytest.mark.parametrize(
+    "elements",
+    [
+        "sub-001",
+        ["sub-001"],
+        ["sub-001", "sub-002"],
+    ],
+)
```

Comment:
Can you add elements that are tuples? Like `[("sub-001", "ses-001"),("sub-001", "ses-002")]`

Reply:
Done.